### ISSUE TITLE
Train custom dataset

### ISSUE THREAD

**Issue by blueclowd** [ruizengalways/PFNet#1]
Hi @ruizengalways 

Thanks for sharing the code. May I know if there is any tutorial showing how to train a custom dataset apart from COCO? Thanks

**Comment by ruizengalways** [comment]
You can use next() to iterate the dataset generator to know the details inside. I think it is not too hard to extend the generator to other datasets

**Comment by SuperDetected** [comment]
Excuse me!
Thanks for your codes.When I use your codes to train in the dataset COCO, the loss is always at 11.xx, and the MACE is always at 24.xx.They never go down.Could you tell me if there are something to adjust?

**Comment by ruizengalways** [comment]
> Excuse me!
> Thanks for your codes.When I use your codes to train in the dataset COCO, the loss is always at 11.xx, and the MACE is always at 24.xx.They never go down.Could you tell me if there are something to adjust?

How many epochs have you already trained? Could you please give me more details about your training?

**Comment by SuperDetected** [comment]
Thanks for your reply,I trained the dataset just with the epochs you have set in the codes:


And the only thing I changed is the IMAGES_PER_GPU,the original value is 32 and&nbsp;I set&nbsp;it 2&nbsp;like this:


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Rui Zeng"<notifications@github.com&gt;;
发送时间:&nbsp;2020年5月27日(星期三) 上午10:03
收件人:&nbsp;"ruizengalways/PFNet"<PFNet@noreply.github.com&gt;;
抄送:&nbsp;"知足~奔跑"<2563163858@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [ruizengalways/PFNet] Train custom dataset (#1)





  
Excuse me!
 Thanks for your codes.When I use your codes to train in the dataset COCO, the loss is always at 11.xx, and the MACE is always at 24.xx.They never go down.Could you tell me if there are something to adjust?
  
How many epochs have you already trained? Could you please give me more details about your training?

**Comment by ruizengalways** [comment]
Increase the batch size and decrease the learning rate



> Thanks for your reply,I trained the dataset just with the epochs you have set in the codes: And the only thing I changed is the IMAGES_PER_GPU,the original value is 32 and&nbsp;I set&nbsp;it 2&nbsp;like this:
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人:&nbsp;"Rui Zeng"<notifications@github.com&gt;; 发送时间:&nbsp;2020年5月27日(星期三) 上午10:03 收件人:&nbsp;"ruizengalways/PFNet"<PFNet@noreply.github.com&gt;; 抄送:&nbsp;"知足~奔跑"<2563163858@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题:&nbsp;Re: [ruizengalways/PFNet] Train custom dataset (#1) Excuse me! Thanks for your codes.When I use your codes to train in the dataset COCO, the loss is always at 11.xx, and the MACE is always at 24.xx.They never go down.Could you tell me if there are something to adjust? How many epochs have you already trained? Could you please give me more details about your training? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by SuperDetected** [comment]
OK, I will have a try,thanks.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Rui Zeng"<notifications@github.com&gt;;
发送时间:&nbsp;2020年5月27日(星期三) 下午3:07
收件人:&nbsp;"ruizengalways/PFNet"<PFNet@noreply.github.com&gt;;
抄送:&nbsp;"知足~奔跑"<2563163858@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [ruizengalways/PFNet] Train custom dataset (#1)





 
Increase the batch size and decrease the learning rate
  
Thanks for your reply,I trained the dataset just with the epochs you have set in the codes: And the only thing I changed is the IMAGES_PER_GPU,the original value is 32 and&nbsp;I set&nbsp;it 2&nbsp;like this:
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人:&nbsp;"Rui Zeng"<notifications@github.com&gt;; 发送时间:&nbsp;2020年5月27日(星期三) 上午10:03 收件人:&nbsp;"ruizengalways/PFNet"<PFNet@noreply.github.com&gt;; 抄送:&nbsp;"知足~奔跑"<2563163858@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; 主题:&nbsp;Re: [ruizengalways/PFNet] Train custom dataset (#1) Excuse me! Thanks for your codes.When I use your codes to train in the dataset COCO, the loss is always at 11.xx, and the MACE is always at 24.xx.They never go down.Could you tell me if there are something to adjust? How many epochs have you already trained? Could you please give me more details about your training? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
